packages/evo/src/agent-plugin/index.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Args_ask,
+  Args_speak,
+  Module,
+  manifest
+} from "./types";
+
+import { Logger } from "@evo-ninja/agent-utils";
+import { PluginFactory, PluginPackage } from "@polywrap/plugin-js";
+
+export interface AgentPluginConfig {
+  logger: Logger;
+}
+
+export class AgentPlugin extends Module<AgentPluginConfig> {
+  private _logger: Logger;
+
+  constructor(config: AgentPluginConfig) {
+    super(config);
+    this._logger = this.config.logger;
+  }
+
+  public speak(args: Args_speak): string {
+    this._logger.success(args.message);
+    return "User has been informed! If you think you've achieved the goal, execute onGoalAchieved.\nIf you think you've failed, execute onGoalFailed.";
```

Comment:
unrelated NIT: we should move this to prompts.ts

Reply:
Noted, will fix in the morning.